documentation/api/man/man3/HttpRequestBuilder.3
.TH "HttpRequestBuilder" 3 "Sat Nov 12 2016" "PHP Web Toolkit 1.0.4 Alpha (phpwebtk)" \" -*- nroff -*-
.ad l
.nh
.SH NAME
HttpRequestBuilder \- 
.SH SYNOPSIS
.br
.PP
.PP
Inherits \fBRequestBuilder\fP\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fB__construct\fP ()"
.br
.ti -1c
.RI "\fBBuildHttpAccept\fP ()"
.br
.ti -1c
.RI "\fBBuildHttpAcceptCharset\fP ()"
.br
.ti -1c
.RI "\fBBuildHttpAcceptEncoding\fP ()"
.br
.ti -1c
.RI "\fBBuildHttpAcceptLanguage\fP ()"
.br
.ti -1c
.RI "\fBBuildHttpConnection\fP ()"
.br
.ti -1c
.RI "\fBBuildHttpGet\fP ()"
.br
.ti -1c
.RI "\fBBuildHttpHost\fP ()"
.br
.ti -1c
.RI "\fBBuildHttpPost\fP ()"
.br
.ti -1c
.RI "\fBBuildHttpReferer\fP ()"
.br
.ti -1c
.RI "\fBBuildHttpUserAgent\fP ()"
.br
.ti -1c
.RI "\fBBuildQueryString\fP ()"
.br
.ti -1c
.RI "\fBBuildRemoteAddress\fP ()"
.br
.ti -1c
.RI "\fBBuildRemoteHost\fP ()"
.br
.ti -1c
.RI "\fBBuildRemotePort\fP ()"
.br
.ti -1c
.RI "\fBBuildRemoteProxyAddr\fP ()"
.br
.ti -1c
.RI "\fBBuildRemoteProxyHost\fP ()"
.br
.ti -1c
.RI "\fBBuildRequestMethod\fP ()"
.br
.ti -1c
.RI "\fBBuildRequestUri\fP ()"
.br
.ti -1c
.RI "\fBBuildServerProtocol\fP ()"
.br
.ti -1c
.RI "\fBGetRequest\fP ()"
.br
.in -1c
.SS "Static Public Member Functions"

.in +1c
.ti -1c
.RI "\fBGetInstance\fP ()"
.br
.in -1c
.SH "Constructor & Destructor Documentation"
.PP 
.SS "HttpRequestBuilder::__construct ()"
function __construct
.PP
This method is executed when an object is instantiated from this class\&. Preprocessing can be done here before the object is put into service\&.
.PP
public 
.SH "Member Function Documentation"
.PP 
.SS "HttpRequestBuilder::BuildHttpAccept ()"
function BuildHttpAccept
.PP
This method sets the contents of the Accept: header from the current request, if there is one\&.
.PP
public 
.SS "HttpRequestBuilder::BuildHttpAcceptCharset ()"
function BuildHttpAcceptCharset
.PP
This method sets the contents of the Accept-Charset: header from the current request, if there is one\&.
.PP
public 
.SS "HttpRequestBuilder::BuildHttpAcceptEncoding ()"
function BuildHttpAcceptEncoding
.PP
This method sets the contents of the Accept-Encoding: header from the current request, if there is one\&.
.PP
public 
.SS "HttpRequestBuilder::BuildHttpAcceptLanguage ()"
function BuildHttpAcceptLanguage
.PP
This method sets the contents of the Accept-Language: header from the current request, if there is one\&.
.PP
public 
.SS "HttpRequestBuilder::BuildHttpConnection ()"
function BuildHttpConnection
.PP
This method sets the contents of the Connection: header from the current request, if there is one\&.
.PP
public 
.SS "HttpRequestBuilder::BuildHttpGet ()"
function BuildHttpGet
.PP
This method sets variables provided to the script via HTTP GET\&.
.PP
public 
.SS "HttpRequestBuilder::BuildHttpHost ()"
function BuildHttpHost
.PP
This method sets the contents of the Host: header from the current request, if there is one\&.
.PP
public 
.SS "HttpRequestBuilder::BuildHttpPost ()"
function BuildHttpPost
.PP
This method sets variables provided to the script via HTTP POST\&.
.PP
public 
.SS "HttpRequestBuilder::BuildHttpReferer ()"
function BuildHttpReferer
.PP
This method sets the address of the page (if any) which referred the user agent to the current page\&. This is set by the user agent\&. Not all user agents will set this, and some provide the ability to modify HTTP_REFERER as a feature\&. In short, it cannot really be trusted\&.
.PP
public 
.SS "HttpRequestBuilder::BuildHttpUserAgent ()"
function BuildHttpUserAgent
.PP
This method sets the contents of the User-Agent: header from the current request, if there is one\&. This is a string denoting the user agent being which is accessing the page\&.
.PP
public 
.SS "HttpRequestBuilder::BuildQueryString ()"
function BuildQueryString
.PP
This method sets the query string, if any, via which the page was accessed\&.
.PP
public 
.SS "HttpRequestBuilder::BuildRemoteAddress ()"
function BuildRemoteAddress
.PP
This method sets the IP address from which the user is viewing the current page\&.
.PP
public 
.SS "HttpRequestBuilder::BuildRemoteHost ()"
function BuildRemoteHost
.PP
This method sets the Host name from which the user is viewing the current page\&. The reverse dns lookup is based off the REMOTE_ADDR of the user\&.
.PP
public 
.SS "HttpRequestBuilder::BuildRemotePort ()"
function BuildRemotePort
.PP
This method sets the port being used on the user's machine to communicate with the web server\&.
.PP
public 
.SS "HttpRequestBuilder::BuildRemoteProxyAddr ()"
function BuildRemoteProxyAddr
.PP
This method sets the proxy IP address from which the user is being forwarded\&.
.PP
public 
.SS "HttpRequestBuilder::BuildRemoteProxyHost ()"
function BuildRemoteProxyHost
.PP
This method sets the Host name from which the user is being forwarded\&. The reverse dns lookup is based off the REMOTE_PROXY_ADDR of the user\&.
.PP
public 
.SS "HttpRequestBuilder::BuildRequestMethod ()"
function BuildRequestMethod
.PP
This method sets the request method that was used to access the page\&.
.PP
public 
.SS "HttpRequestBuilder::BuildRequestUri ()"
function BuildRequestUri
.PP
This method retreives the URI which was given in order to access this page\&.
.PP
public 
.SS "HttpRequestBuilder::BuildServerProtocol ()"
function BuildServerProtocol
.PP
This method sets the name and revision of the information protocol via which the page was requested\&.
.PP
public 
.SS "HttpRequestBuilder::GetInstance ()\fC [static]\fP"
function GetInstance
.PP
This method instantiates a new object from this class; more specifically, it's a singleton instance\&.
.PP
public 
.PP
\fBReturns:\fP
.RS 4
\fBHttpRequestBuilder\fP object instance 
.RE
.PP

.SS "HttpRequestBuilder::GetRequest ()"
function GetRequest
.PP
This method returns the \fBRequest\fP object to the calling method\&.
.PP
public 
.PP
\fBReturns:\fP
.RS 4
\fBRequest\fP object instance 
.RE
.PP


.SH "Author"
.PP 
Generated automatically by Doxygen for PHP Web Toolkit 1\&.0\&.4 Alpha (phpwebtk) from the source code\&.


documentation/CHANGELOG.md
12 Nov 2016

* General:
   * Formatted all the code using Zend Eclipse PDT

* Documentation:
   * phpDocumentor for PHP 5 will no longer be used
   * doxygen is now being used to create new API docs

* SampleView:
   * Used new method of turning off smarty cache
   * Removed obselete code left over from Smarty2
   * Added support for Smarty3 security policy
   * Moved all CSS into their own sub-directories

* View:
   * Removed extraneous slash in URI for HTTP header

* XmlDaoFactory:
   * Using new function name and init for DOMDocument


httprequestbuilder.class.php
<?php
/**
 * $Id$
 * PHP Web Toolkit Version 1.0.4 Alpha
 *
 * @package phpwebtk
 */
/**
 * class HttpRequestBuilder
 *
 * This class constructs the parts of the Request by implementing the
 * RequestBuilder interface, defines and keeps track of the
 * representation it creates and provides an interface for retrieving
 * the Request object.
 *
 * @author Brian Bisaillon <bisailb@myprivacy.ca>
 * @copyright Copyright (C) 2004-2016 by Brian Bisaillon
 * @package phpwebtk
 * @subpackage http
 */
class HttpRequestBuilder extends RequestBuilder {
	// Private members
	private static $HttpRequestBuilder = null;
	private $Request;
	// Private properties
	private $ipAddress;
	private $ipAddresses;
	private $proxyAddress;
	/**
	 * function __construct
	 *
	 * This method is executed when an object is instantiated from this
	 * class. Preprocessing can be done here before the object is put
	 * into service.
	 *
	 * @access public
	 */
	public function __construct() {
		$this->Request = new Request ();
		$this->GetHttpClientIPAddress ();
	}
	/**
	 * function GetInstance
	 *
	 * This method instantiates a new object from this class; more
	 * specifically, it's a singleton instance.
	 *
	 * @access public
	 * @return HttpRequestBuilder object instance
	 * @static
	 *
	 */
	public function GetInstance() {
		if (empty ( HttpRequestBuilder::$HttpRequestBuilder )) {
			HttpRequestBuilder::$HttpRequestBuilder = new HttpRequestBuilder ();
		}
		return (HttpRequestBuilder::$HttpRequestBuilder);
	}
	/**
	 * function GetHttpClientIPAddress
	 *
	 * This method sets the end user's IP address and proxy address if
	 * there is one.
	 *
	 * @access private
	 */
	private function GetHttpClientIPAddress() {
		if (isset ( $_SERVER ['HTTP_X_FORWARDED_FOR'] ) || isset ( $_SERVER ['HTTP_CLIENT_IP'] ) || isset ( $_SERVER ['REMOTE_ADDR'] )) {
			if (isset ( $_SERVER ['HTTP_X_FORWARDED_FOR'] )) {
				$this->ipAddress = $_SERVER ['HTTP_X_FORWARDED_FOR'];
				$this->proxyAddress = $_SERVER ['REMOTE_ADDR'];
			} else if (isset ( $_SERVER ['HTTP_CLIENT_IP'] )) {
				$this->ipAddress = $_SERVER ['HTTP_CLIENT_IP'];
			} else if (isset ( $_SERVER ['REMOTE_ADDR'] )) {
				$this->ipAddress = $_SERVER ['REMOTE_ADDR'];
			}
		} else {
			if (getenv ( 'HTTP_X_FORWARDED_FOR' )) {
				$this->ipAddress = getenv ( 'HTTP_X_FORWARDED_FOR' );
				$this->proxyAddress = getenv ( 'REMOTE_ADDR' );
			} else if (getenv ( 'HTTP_CLIENT_IP' )) {
				$this->ipAddress = getenv ( 'HTTP_CLIENT_IP' );
			} else {
				$this->ipAddress = getenv ( 'REMOTE_ADDR' );
			}
		}
		if (strstr ( $this->ipAddress, ',' )) {
			$this->ipAddresses = explode ( ',', $this->ipAddress );
			$this->ipAddress = $this->ipAddresses [0];
		}
	}
	/**
	 * function BuildHttpAccept
	 *
	 * This method sets the contents of the Accept: header from the
	 * current request, if there is one.
	 *
	 * @access public
	 */
	public function BuildHttpAccept() {
		if (isset ( $_SERVER ['HTTP_ACCEPT'] )) {
			$this->Request->SetProperty ( 'HTTP_ACCEPT', $_SERVER ['HTTP_ACCEPT'] );
		}
	}
	/**
	 * function BuildHttpAcceptCharset
	 *
	 * This method sets the contents of the Accept-Charset: header from
	 * the current request, if there is one.
	 *
	 * @access public
	 */
	public function BuildHttpAcceptCharset() {
		if (isset ( $_SERVER ['HTTP_ACCEPT_CHARSET'] )) {
			$this->Request->SetProperty ( 'HTTP_ACCEPT_CHARSET', $_SERVER ['HTTP_ACCEPT_CHARSET'] );
		}
	}
	/**
	 * function BuildHttpAcceptEncoding
	 *
	 * This method sets the contents of the Accept-Encoding: header from
	 * the current request, if there is one.
	 *
	 * @access public
	 */
	public function BuildHttpAcceptEncoding() {
		if (isset ( $_SERVER ['HTTP_ACCEPT_ENCODING'] )) {
			$this->Request->SetProperty ( 'HTTP_ACCEPT_ENCODING', $_SERVER ['HTTP_ACCEPT_ENCODING'] );
		}
	}
	/**
	 * function BuildHttpAcceptLanguage
	 *
	 * This method sets the contents of the Accept-Language: header from
	 * the current request, if there is one.
	 *
	 * @access public
	 */
	public function BuildHttpAcceptLanguage() {
		if (isset ( $_SERVER ['HTTP_ACCEPT_LANGUAGE'] )) {
			$this->Request->SetProperty ( 'HTTP_ACCEPT_LANGUAGE', $_SERVER ['HTTP_ACCEPT_LANGUAGE'] );
		}
	}
	/**
	 * function BuildHttpConnection
	 *
	 * This method sets the contents of the Connection: header from the
	 * current request, if there is one.
	 *
	 * @access public
	 */
	public function BuildHttpConnection() {
		if (isset ( $_SERVER ['HTTP_CONNECTION'] )) {
			$this->Request->SetProperty ( 'HTTP_CONNECTION', $_SERVER ['HTTP_CONNECTION'] );
		}
	}
	/**
	 * function BuildHttpGet
	 *
	 * This method sets variables provided to the script via HTTP GET.
	 *
	 * @access public
	 */
	public function BuildHttpGet() {
		if (isset ( $_GET )) {
			$this->Request->SetProperty ( 'HTTP_GET', $_GET );
		}
	}
	/**
	 * function BuildHttpHost
	 *
	 * This method sets the contents of the Host: header from the
	 * current request, if there is one.
	 *
	 * @access public
	 */
	public function BuildHttpHost() {
		if (isset ( $_SERVER ['HTTP_HOST'] )) {
			$this->Request->SetProperty ( 'HTTP_HOST', $_SERVER ['HTTP_HOST'] );
		}
	}
	/**
	 * function BuildHttpPost
	 *
	 * This method sets variables provided to the script via HTTP POST.
	 *
	 * @access public
	 */
	public function BuildHttpPost() {
		if (isset ( $_POST )) {
			$this->Request->SetProperty ( 'HTTP_POST', $_POST );
		}
	}
	/**
	 * function BuildHttpReferer
	 *
	 * This method sets the address of the page (if any) which referred
	 * the user agent to the current page. This is set by the user
	 * agent. Not all user agents will set this, and some provide the
	 * ability to modify HTTP_REFERER as a feature. In short, it cannot
	 * really be trusted.
	 *
	 * @access public
	 */
	public function BuildHttpReferer() {
		if (isset ( $_SERVER ['HTTP_REFERER'] )) {
			$this->Request->SetProperty ( 'HTTP_REFERER', $_SERVER ['HTTP_REFERER'] );
		}
	}
	/**
	 * function BuildHttpUserAgent
	 *
	 * This method sets the contents of the User-Agent: header from the
	 * current request, if there is one. This is a string denoting the
	 * user agent being which is accessing the page.
	 *
	 * @access public
	 */
	public function BuildHttpUserAgent() {
		if (isset ( $_SERVER ['HTTP_USER_AGENT'] )) {
			$this->Request->SetProperty ( 'HTTP_USER_AGENT', $_SERVER ['HTTP_USER_AGENT'] );
		}
	}
	/**
	 * function BuildQueryString
	 *
	 * This method sets the query string, if any, via which the page was
	 * accessed.
	 *
	 * @access public
	 */
	public function BuildQueryString() {
		if (isset ( $_SERVER ['QUERY_STRING'] )) {
			$this->Request->SetProperty ( 'QUERY_STRING', $_SERVER ['QUERY_STRING'] );
		}
	}
	/**
	 * function BuildRemoteAddress
	 *
	 * This method sets the IP address from which the user is viewing
	 * the current page.
	 *
	 * @access public
	 */
	public function BuildRemoteAddress() {
		if (TRUE !== empty ( $this->ipAddress )) {
			$this->Request->SetProperty ( 'REMOTE_ADDR', $this->ipAddress );
		}
	}
	/**
	 * function BuildRemoteHost
	 *
	 * This method sets the Host name from which the user is viewing the
	 * current page. The reverse dns lookup is based off the REMOTE_ADDR
	 * of the user.
	 *
	 * @access public
	 */
	public function BuildRemoteHost() {
		if (TRUE !== empty ( $this->ipAddress )) {
			$this->Request->SetProperty ( 'REMOTE_HOST', gethostbyaddr ( $this->ipAddress ) );
		}
	}
	/**
	 * function BuildRemotePort
	 *
	 * This method sets the port being used on the user's machine to
	 * communicate with the web server.
	 *
	 * @access public
	 */
	public function BuildRemotePort() {
		if (isset ( $_SERVER ['REMOTE_PORT'] )) {
			$this->Request->SetProperty ( 'REMOTE_PORT', $_SERVER ['REMOTE_PORT'] );
		}
	}
	/**
	 * function BuildRemoteProxyAddr
	 *
	 * This method sets the proxy IP address from which the user is
	 * being forwarded.
	 *
	 * @access public
	 */
	public function BuildRemoteProxyAddr() {
		if (TRUE !== empty ( $this->proxyAddress )) {
			$this->Request->SetProperty ( 'REMOTE_PROXY_ADDR', $this->proxyAddress );
		}
	}
	/**
	 * function BuildRemoteProxyHost
	 *
	 * This method sets the Host name from which the user is being
	 * forwarded. The reverse dns lookup is based off the
	 * REMOTE_PROXY_ADDR of the user.
	 *
	 * @access public
	 */
	public function BuildRemoteProxyHost() {
		if (TRUE !== empty ( $this->proxyAddress )) {
			$this->Request->SetProperty ( 'REMOTE_PROXY_HOST', gethostbyaddr ( $this->proxyAddress ) );
		}
	}
	/**
	 * function BuildRequestMethod
	 *
	 * This method sets the request method that was used to access the
	 * page.
	 *
	 * @access public
	 */
	public function BuildRequestMethod() {
		if (isset ( $_SERVER ['REQUEST_METHOD'] )) {
			$this->Request->SetProperty ( 'REQUEST_METHOD', $_SERVER ['REQUEST_METHOD'] );
		}
	}
	/**
	 * function BuildRequestUri
	 *
	 * This method retreives the URI which was given in order to access
	 * this page.
	 *
	 * @access public
	 */
	public function BuildRequestUri() {
		if (isset ( $_SERVER ['REQUEST_URI'] )) {
			$this->Request->SetProperty ( 'REQUEST_URI', $_SERVER ['SCRIPT_NAME'] . '?' . $_SERVER ['QUERY_STRING'] );
		} else if (isset ( $_SERVER ['REQUEST_URI'] )) {
			$this->Request->SetProperty ( 'REQUEST_URI', $_SERVER ['REQUEST_URI'] );
		}
	}
	/**
	 * function BuildServerProtocol
	 *
	 * This method sets the name and revision of the information
	 * protocol via which the page was requested.
	 *
	 * @access public
	 */
	public function BuildServerProtocol() {
		if (isset ( $_SERVER ['SERVER_PROTOCOL'] )) {
			$this->Request->SetProperty ( 'SERVER_PROTOCOL', $_SERVER ['SERVER_PROTOCOL'] );
		}
	}
	/**
	 * function GetRequest
	 *
	 * This method returns the Request object to the calling method.
	 *
	 * @access public
	 * @return Request object instance
	 */
	public function GetRequest() {
		return ($this->Request);
	}
}
?>
